application/src/test/java/uk/nhs/hee/tis/revalidation/validator/TraineeRecommendationRecordDTOValidatorTest.java

```diff
@@ -0,0 +1,235 @@
+ * DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package uk.nhs.hee.tis.revalidation.validator;
+
+import static java.time.LocalDate.now;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.when;
+import static uk.nhs.hee.tis.revalidation.entity.RecommendationType.DEFER;
+
+import com.github.javafaker.Faker;
+import java.time.LocalDate;
+import java.util.List;
+import java.util.Optional;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.validation.BeanPropertyBindingResult;
+import org.springframework.validation.Errors;
+import uk.nhs.hee.tis.revalidation.dto.TraineeRecommendationRecordDto;
+import uk.nhs.hee.tis.revalidation.entity.DoctorsForDB;
+import uk.nhs.hee.tis.revalidation.exception.RecommendationException;
+import uk.nhs.hee.tis.revalidation.repository.DoctorsForDBRepository;
+
+@ExtendWith(MockitoExtension.class)
+class TraineeRecommendationRecordDTOValidatorTest {
+
+  private final static LocalDate today = now();
+  private final Faker faker = new Faker();
+  private final LocalDate gmcSubmissionDate = now();
+  private final LocalDate deferralDate = now();
+  private final String gmcId = faker.number().digits(7);
+  private final String deferralReason1 = "1";
+  private final String deferralSubReason1 = "1";
+  private static final String DOCTOR_NOT_FOUND_MESSAGE = "Doctor %s does not exist!";
+  @Mock
+  private DoctorsForDBRepository doctorsForDBRepository;
+  @InjectMocks
+  private TraineeRecommendationRecordDTOValidator validatorUnderTest;
+  private DoctorsForDB doctorsForDB;
+
+
+  @Test
+  void shouldValidateWhenGmcNumberOrRecommendationTypeEmptyOrNullInRecommendationRequest() {
+
+    //Given
+    final var recordDTO = TraineeRecommendationRecordDto.builder()
+        .gmcNumber("")
+        .recommendationType("")
+        .comments(List.of())
+        .build();
+
+    Errors errors = new BeanPropertyBindingResult(recordDTO, "nullGmcOrRecommendationType");
+
+    //When
+    validatorUnderTest.validate(recordDTO, errors);
+
+    //Then
+    assertThat(true, is(errors.hasErrors()));
+    assertThat("nullGmcOrRecommendationType", is(errors.getObjectName()));
+    assertThat("GmcNumber", is(errors.getAllErrors().get(0).getCode()));
+    assertThat("Gmc Number can't be empty or null",
+        is(errors.getAllErrors().get(0).getDefaultMessage()));
+    assertThat("RecommendationType", is(errors.getAllErrors().get(1).getCode()));
+    assertThat("Recommendation type can't be empty or null",
+        is(errors.getAllErrors().get(1).getDefaultMessage()));
```

Comment:
This seems a bit sensitive to the ordering that errors are placed & stored in the BindingResult.  Using more flexible list/collection matchers e.g. "hasItems()"

Reply:
Need your help here